tests/integration/test_gerrit.py

```diff
@@ -0,0 +1,119 @@
+                ('git push --porcelain --progress {} HEAD:refs/for/{}'.format(self.service.name, branch), b'', '\n'.join([
+                    'Counting objects: 6, done.',
+                    'Delta compression using up to 2 threads.',
+                    'Compressing objects: 100% (6/6), done.',
+                    'Writing objects: 100% (6/6), 515 bytes | 0 bytes/s, done.',
+                    'Total 6 (delta 4), reused 0 (delta 0)',
+                    'remote: Resolving deltas: 100% (4/4)',
+                    'remote: Processing changes: new: 1, refs: 1, done',
+                    'remote: ',
+                    'remote: New Changes:',
+                    'remote:   https://{}/391808 One more test'.format(self.service.fqdn),
+                    'remote: ',
+                    'To {}'.format(local_slug),
+                    '*	HEAD:refs/for/{}	[new branch]'.format(branch),
+                    'Done'
+                ]).encode('utf-8'), 0)
+            ])
+            with self.recorder.use_cassette(self._make_cassette_name()):
+                self.service.connect()
+                self.service.request_create(namespace, repository, branch)
+
+    def action_request_fetch(self, namespace, repository, request, ref):
+        local_slug = self.service.format_path(namespace=namespace, repository=repository, rw=True)
+        self.repository.create_remote('all', url=local_slug)
+        self.repository.create_remote(self.service.name, url=local_slug)
+
+        with self.mockup_git(namespace, repository):
+            self.set_mock_popen_commands([
+                ('git version', b'git version 2.7.4', b'', 0),
+                ('git fetch --progress --update-head-ok -v gerrithub {}'.format(ref), b'', '\n'.join([
+                    'From {}'.format(local_slug),
+                    '* branch            {} -> FETCH_HEAD'.format(ref)
+                ]).encode('utf-8'), 0),
+                ('git checkout FETCH_HEAD', b'', '\n'.join([
+                    'Note: checking out \'FETCH_HEAD\'',
+                    '',
+                    'You are in \'detached HEAD\' state. You can look around, make experimental',
+                    'changes and commit them, and you can discard any commits you make in this',
+                    'state without impacting any branches by performing another checkout.',
+                    '',
+                    'If you want to create a new branch to retain commits you create, you may',
+                    'do so (now or later) by using -b with the checkout command again. Example:',
+                    '',
+                    '  git checkout -b <new-branch-name>',
+                    '',
+                    'HEAD is now at 5bb7935... One more test',
+                ]).encode('utf-8'), 0)
+            ])
+            with self.recorder.use_cassette(self._make_cassette_name()):
+                self.service.connect()
+                self.service.request_fetch(namespace, repository, request)
```

Comment:
I see no good reason to overload `action_request_fetch`, especially if you follow the way it's working for the other services (fetching the request as a new ref, give the name of the ref, but not switch to it).

If you really like the behaviour of fetch + checkout, then we could consider creating a `request pull` option 😉